graphql/patch_resolver.go

```diff
@@ -162,18 +173,11 @@ func (r *patchResolver) TaskCount(ctx context.Context, obj *restModel.APIPatch)
 
 // TaskStatuses is the resolver for the taskStatuses field.
 func (r *patchResolver) TaskStatuses(ctx context.Context, obj *restModel.APIPatch) ([]string, error) {
-	defaultSort := []task.TasksSortOrder{
-		{Key: task.DisplayNameKey, Order: 1},
-	}
-	opts := task.GetTasksByVersionOptions{
-		Sorts:            defaultSort,
-		IncludeBaseTasks: false,
-	}
-	tasks, _, err := task.GetTasksByVersion(ctx, *obj.Id, opts)
+	statuses, err := task.GetTaskStatusesByVersion(ctx, utility.FromStringPtr(obj.Id))
```

Comment:
If we're trying to make the patch/version resolver follow the same logic, would it be worth consolidating this + BaseTaskStatsuses into a helper function that the patch + version resolvers can share?

Reply:
I created a helper function to fetch base versions. But I didn't go as far as making one specifically for statuses. Since patches don't always have an associated version so it's okay if this field returns nil. But if we can't find a version, we likely have a bug.